components/dashboard/src/menu/Menu.tsx

```diff
@@ -0,0 +1,221 @@
+import gitpodIcon from "../icons/gitpod.svg";
+import { getGitpodService, gitpodHostUrl } from "../service/service";
+import { useCurrentUser } from "../user-context";
+import { useBillingModeForCurrentTeam, useCurrentTeam, useTeamMemberInfos } from "../teams/teams-context";
+import ContextMenu from "../components/ContextMenu";
+import Separator from "../components/Separator";
+import PillMenuItem from "../components/PillMenuItem";
+import { getTeamSettingsMenu } from "../teams/TeamSettings";
+import { PaymentContext } from "../payment-context";
+import FeedbackFormModal from "../feedback-form/FeedbackModal";
+import { inResource, isGitpodIo } from "../utils";
+import { BillingMode } from "@gitpod/gitpod-protocol/lib/billing-mode";
+import { FeatureFlagContext } from "../contexts/FeatureFlagContext";
+import OrganizationSelector from "./OrganizationSelector";
+
+interface Entry {
+    title: string;
+    link: string;
+    alternatives?: string[];
+}
+
+export default function Menu() {
+    const user = useCurrentUser();
+    const team = useCurrentTeam();
+    const location = useLocation();
+    const teamBillingMode = useBillingModeForCurrentTeam();
+    const { showUsageView, oidcServiceEnabled } = useContext(FeatureFlagContext);
+    const { setCurrency, setIsStudent, setIsChargebeeCustomer } = useContext(PaymentContext);
+    const [userBillingMode, setUserBillingMode] = useState<BillingMode | undefined>(undefined);
+    const [isFeedbackFormVisible, setFeedbackFormVisible] = useState<boolean>(false);
+    const teamMembers = useTeamMemberInfos();
+
+    useEffect(() => {
+        getGitpodService().server.getBillingModeForUser().then(setUserBillingMode);
+    }, []);
+
+    function isSelected(entry: Entry, location: Location<any>) {
+        const all = [entry.link, ...(entry.alternatives || [])].map((l) => l.toLowerCase());
+        const path = location.pathname.toLowerCase();
+        return all.some((n) => n === path || n + "/" === path);
+    }
+
+    // Hide most of the top menu when in a full-page form.
+    const isMinimalUI = inResource(location.pathname, ["new", "orgs/new", "open"]);
+
+    useEffect(() => {
+        const { server } = getGitpodService();
+        Promise.all([
+            server.getClientRegion().then((v) => () => {
+                // @ts-ignore
+                setCurrency(countries[v]?.currency === "EUR" ? "EUR" : "USD");
+            }),
+            server.isStudent().then((v) => () => setIsStudent(v)),
+            server.isChargebeeCustomer().then((v) => () => setIsChargebeeCustomer(v)),
+        ]).then((setters) => setters.forEach((s) => s()));
+    }, [setCurrency, setIsChargebeeCustomer, setIsStudent]);
+
+    const leftMenu = useMemo(() => {
+        const leftMenu: Entry[] = [
+            {
+                title: "Workspaces",
+                link: "/workspaces",
+                alternatives: ["/"],
+            },
+            {
+                title: "Projects",
```

Comment:
**praise**: Having the workspaces, projects, etc preselected when changing org feels magical and something that has been bothering me for a while. Thanks for making this change! ✨